### DISCUSSION TITLE
Impersonation, aka. "Log in as"

### DISCUSSION THREAD

**Discussion by antonmihaylov** [zitadel/zitadel#5895] [Ideas]
I'm not sure if this is already possible in some way with the existing API, I would love a pointer to the docs if it is.

But for any larger app it would be very important to be able to impersonate users, mostly for remote support, but also for testing and other scenarios.

One concrete idea for the mechanism is to allow any custom backend, for example authenticated with a Service account as a IAM owner, to be able to call an API and get an access token for any user.

**Comment by hifabienne** [comment]
Hei @antonmihaylov 
Thanks for opening up the discussion.
Impersonation is currently not possible.
I think it is something that could be interesting for multiple customers.
If we see that the need is high enough we will have a look at it.

**Reply by livio-a** [reply]
hey @antonmihaylov 

There were multiple requests for impersonation or similar related topics. We're therefore trying to collect it in a single discussion.
I noted your cases in this [discussion](https://github.com/zitadel/zitadel/discussions/7070), where we also try to find the best solution for the implementation.
